collectors/apps.plugin/apps_plugin.c

```diff
@@ -4134,6 +4174,511 @@ static int check_capabilities() {
+#ifndef __FreeBSD__
+                                "\"lreads\", "
+#endif
+                                "\"preads\" ]"
+                "\n      },"
+                "\n      \"writes\": {"
+                "\n         \"name\":\"I/O Writes\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ "
+#ifndef __FreeBSD__
+                                "\"lwrites\", "
+#endif
+                                "\"pwrites\" ]"
+                "\n      },"
+#ifndef __FreeBSD__
+                "\n      \"logical_io\": {"
+                "\n         \"name\":\"Logical I/O\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"lreads\", \"lwrites\" ]"
+                "\n      },"
+#endif
+                "\n      \"physical_io\": {"
+                "\n         \"name\":\"Physical I/O\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"preads\", \"pwrites\" ]"
+                "\n      },"
+                "\n      \"io_calls\": {"
+                "\n         \"name\":\"I/O Calls\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"readcalls\", \"writecalls\" ]"
+                "\n      },"
+                "\n      \"minflt\": {"
+                "\n         \"name\":\"Minor Page Faults\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"minflt\", \"cminflt\" ]"
+                "\n      },"
+                "\n      \"majflt\": {"
+                "\n         \"name\":\"Major Page Faults\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"majflt\", \"cmajflt\" ]"
+                "\n      },"
+                "\n      \"threads\": {"
+                "\n         \"name\":\"Threads\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"threads\" ]"
+                "\n      },"
+                "\n      \"procs\": {"
+                "\n         \"name\":\"Processes\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"procs\" ]"
+                "\n      },"
+                "\n      \"fds\": {"
+                "\n         \"name\":\"File Descriptors\","
+                "\n         \"type\":\"stacked-bar\","
+                "\n         \"columns\": [ \"fds\", \"pipefds\", \"socketfds\", \"inotifyfds\", \"eventfds\", \"timerfds\", \"signalfds\", \"eventpolls\", \"otherfds\" ]"
+                "\n      }"
+                "\n   },"
+                "\n   \"data\":["
+                "\n"
+                );
+    }
+
+    fwrite(buffer_tostring(func_processes_fields), buffer_strlen(func_processes_fields), 1, stdout);
+
+    NETDATA_DOUBLE cpu_divisor = (NETDATA_DOUBLE)(time_factor * RATES_DETAIL / 100);
```

Comment:
Hello @ktsaou ,

A small report from coverity:

```sh
>>>     CID 358247:  Incorrect expression  (UNINTENDED_INTEGER_DIVISION)
>>>     Dividing integer expressions "time_factor * 10000ULL" and "100ULL", and then converting the integer quotient to type "double". Any remainder, or fractional part of the quotient, is ignored.
4493         NETDATA_DOUBLE cpu_divisor = (NETDATA_DOUBLE)(time_factor * RATES_DETAIL / 100);
```

Reply:
I solved it.